### PR TITLE
feat: add dead letter queue samples

### LINKED ISSUE #1178: Pub/Sub: please add DLQ samples for Pub/Sub
Please add samples for 

1. create a subscription with a dead letter policy: `pubsub_dead_letter_create_subscription`
1. update a subscription's dead letter policy: `pubsub_dead_letter_update_subscription`
1. remove a subscription's dead letter policy: `pubsub_dead_letter_remove`
1. show the delivery attempt field in received messages: `pubsub_dead_letter_delivery_attempt`

See existing [Java](https://github.com/googleapis/java-pubsub/tree/master/samples/snippets/src/main/java/pubsub), [Go](https://github.com/GoogleCloudPlatform/golang-samples/pull/1279), [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/3904), [Node](https://github.com/googleapis/nodejs-pubsub/pull/1010) samples.

### PR DESCRIPTION
**Pull request by jdpedrie**
closes #1178.

### CODE REVIEW

File: pubsub/api/src/dead_letter_update_subscription.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/blob/master/pubsub/api/README.md
+ */
+
+namespace Google\Cloud\Samples\PubSub;
+
+# [START pubsub_dead_letter_remove]
```
Comment (anguillanneuf) [inline pubsub/api/src/dead_letter_update_subscription.php:26]: This should be `pubsub_dead_letter_update_subscription`.

File: pubsub/api/src/dead_letter_update_subscription.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ]
+    ]);
+
+    printf(
+        'Subscription %s updated with dead letter topic %s' . PHP_EOL,
+        $subscription->name(),
+        $deadLetterTopic->name()
+    );
+}
+# [END pubsub_dead_letter_remove]
```
Comment (anguillanneuf) [inline pubsub/api/src/dead_letter_update_subscription.php:58]: Ditto.

**Comment by jdpedrie** [comment]
Thanks @anguillanneuf. This is fixed!

**Review by anguillanneuf — APPROVED** [review]

**Review by bshaffer — APPROVED** [review]

File: pubsub/api/pubsub.php [hunk 3]
```diff
@@ -138,6 +138,108 @@
...
+        --subscription my-subscription
+        --dead-letter-topic my-dead-letter-topic</info>
+
+    <info>php %command.full_name% remove --project my-project
+        --topic my-topic</info>
+
+    <info>php %command.full_name% pull --project my-project
+        --topic my-topic --subscription my-subscription</info>
+
+    EOF
```
Comment (bshaffer) [inline pubsub/api/pubsub.php:162]: Getting a kokoro syntax error for PHP 7.2 on this line. This needs to be flush with no indentation
```suggestion
EOF
```

File: pubsub/api/test/DeadLetterPolicyTest.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        self::$deadLetterTopic2 = self::$client->createTopic(self::$deadLetterTopic2Name);
+        self::$subscription = self::$client->subscription(self::$subscriptionName);
+    }
+
+    public static function tearDownAfterClass()
+    {
+         self::$topic->delete();
+         self::$subscription->delete();
+         self::$deadLetterTopic->delete();
+         self::$deadLetterTopic2->delete();
```
Comment (bshaffer) [inline pubsub/api/test/DeadLetterPolicyTest.php:68]: Looks like these four lines contain an extra space in their indentation:
```
   1) pubsub/api/test/DeadLetterPolicyTest.php
      ---------- begin diff ----------
--- Original
+++ New
@@ @@
     {
-         self::$topic->delete();
-         self::$subscription->delete();
-         self::$deadLetterTopic->delete();
-         self::$deadLetterTopic2->delete();
+        self::$topic->delete();
+        self::$subscription->delete();
+        self::$deadLetterTopic->delete();
+        self::$deadLetterTopic2->delete();
     }
```

**Review by bshaffer — CHANGES_REQUESTED** [review]

**Review by bshaffer — APPROVED** [review]

**Pull request merged**